identify_monographs.py

```diff
@@ -0,0 +1,402 @@
+    monograph_forms = _get_alma_monograph_forms(alma_record)
+    normalized_monograph_forms = _normalize_alma_monograph_form_values(monograph_forms)
+    forms["alma"] = normalized_monograph_forms
+    return forms
+
+
+def compare_monograph_forms(alma_record: dict, fm_record: dict) -> dict:
+    """Compares monograph forms between Alma and FileMaker records."""
+    forms = get_monograph_forms(alma_record, fm_record)
+    alma_forms = forms.get("alma", [])
+    fm_forms = forms.get("filemaker", [])
+    common_forms = set(alma_forms).intersection(set(fm_forms))
+    unique_alma_forms = set(alma_forms).difference(set(fm_forms))
+    unique_fm_forms = set(fm_forms).difference(set(alma_forms))
+    comparison = {
+        "MMS ID": alma_record["001"].data,
+        "common": list(common_forms),
+        "unique_alma": list(unique_alma_forms),
+        "unique_fm": list(unique_fm_forms),
+    }
+    return comparison
+
+
+def _write_form_mismatch_report(monograph_records: list, output_file: str) -> None:
+    """Writes a form mismatch report to a CSV file."""
+    comparison_data = []
+    for alma_record, fm_record in monograph_records:
+        comparison = compare_monograph_forms(alma_record, fm_record)
+        comparison_data.append(comparison)
+    # filter to only records with form mismatches
+    comparison_data = [
+        record
+        for record in comparison_data
+        if record["unique_alma"] or record["unique_fm"]
+    ]
+    print(f"Found {len(comparison_data)} records with form mismatches")
+    with open(output_file, "w") as f:
+        writer = csv.DictWriter(
+            f,
+            fieldnames=["MMS ID", "common", "unique_alma", "unique_fm"],
+        )
+        writer.writeheader()
+        writer.writerows(comparison_data)
+
+
+def _write_monograph_records(records: list, output_file: str) -> None:
+    """Writes monograph records to a MARC file."""
+    with open(output_file, "wb") as f:
+        for alma_record, fm_record in records:
+            f.write(alma_record.as_marc())
+
+
+def main():
+    args = _get_args()
+    alma_data_refs = json.load(open(args.alma_holdings_file))
+    fm_data = csv.DictReader(
+        open(args.filemaker_data_file), fieldnames=["inventory_no", "production_type"]
+    )
+    fm_data = [row for row in fm_data]
+    matched_records = _match_records(alma_data_refs, fm_data)
+    print(f"Matched {len(matched_records)} records")
+    full_matched_records = _get_full_bib_data(matched_records, args.alma_bibs_file)
+    print(f"Matched {len(full_matched_records)} full records")
+    monograph_records = []
+    for alma_record, fm_record in full_matched_records:
+        if is_monograph(alma_record, fm_record):
+            monograph_records.append((alma_record, fm_record))
+    print(f"Found {len(monograph_records)} monograph records")
+
+    if args.form_mismatch_report:
+        print(f"Writing form mismatch report to {args.form_mismatch_report}")
```

Comment:
I think print statements like this belong in the method they're associated with, since they're so closely related.